client/server/server.go

```diff
@@ -131,6 +131,12 @@ func (s *Server) Login(_ context.Context, msg *proto.LoginRequest) (*proto.Login
 	s.config = config
 	s.mutex.Unlock()
 
+	if s, _ := state.Status(); s != internal.StatusNeedsLogin && s != internal.StatusLoginFailed {
```

Comment:
This s seems to be shadowing the server variable, so maybe we can rename it to something like status